src/components/SQForm/SQFormMultiValue.js

```diff
@@ -109,8 +110,39 @@ function SQFormMultiValue({
 
   const [inputValue, setInputValue] = React.useState('');
   const [customOptions, setCustomOptions] = React.useState([]);
+  const displayOptions = React.useMemo(() => [...children, ...customOptions], [
+    children,
+    customOptions
+  ]);
+  const previousInitialValue = usePrevious(initialValue);
 
-  const initialValue = [];
+  React.useEffect(() => {
+    // Don't run this effect every time displayOptions changes
+    // This also prevents infinite rerenders
+    if (JSON.stringify(previousInitialValue) === JSON.stringify(initialValue)) {
+      return;
+    }
+
+    const displayValues = displayOptions.reduce((acc, option) => {
+      acc.push(option.value);
+      return acc;
+    }, []);
+
+    const nonOptionValues = initialValue.filter(value => {
+      return !displayValues.includes(value);
+    });
+
+    const newCustomOptions = nonOptionValues.map(newCustomOptionValue => {
+      return {
+        label: newCustomOptionValue,
+        value: newCustomOptionValue
+      };
+    });
```

Comment:
I am finding some of the naming a bit awkward. 
Both 131 and 135 are new custom values, i.e. they are values that are not in the list of predefined options, correct? 

Also you could maybe combine the filter and map into another reduce. Might clean it up a bit and reduce the number of loops. The use case you are building this for I don't think will have a huge set, but other consumers might be glad to cut a loop out.

Reply:
> I am finding some of the naming a bit awkward.
> Both 131 and 135 are new custom values, i.e. they are values that are not in the list of predefined options, correct?
> 
> Also you could maybe combine the filter and map into another reduce. Might clean it up a bit and reduce the number of loops. The use case you are building this for I don't think will have a huge set, but other consumers might be glad to cut a loop out.

I WOULD recommend taking the time to reduce (unintentional pun) the number of iterations. Whenever we step into the realm of an autocomplete, we offer an autocomplete because there's a large number of options. So many, it's much faster to type than to scroll.

We have windowing implemented which is the biggest perf gain, but if we have a dropdown of 10000 options, do we want to perform 20000 iterations (map + filter), or 10000 iterations (reduce)?

Reply:
That's definitely a good call out and an easy fix and implemented.